packages/sync-actions/test/customer-sync.spec.js

```diff
@@ -533,4 +539,96 @@ describe('Actions', () => {
       },
     ])
   })
+
+  test('should build setAuthenticationMode sync action', () => {
+    let before
+    let now
+    let actual
+    let expected
+
+    before = {
+      authenticationMode: 'Password',
+    }
+    now = {
+      authenticationMode: 'ExternalAuth',
+    }
+
+    actual = customerSync.buildActions(now, before)
+
+    expected = [
+      {
+        action: 'setAuthenticationMode',
+        authMode: now.authenticationMode,
+      },
+    ]
+    expect(actual).toEqual(expected)
+
+    before = {
+      authenticationMode: 'ExternalAuth',
+    }
+    now = {
+      authenticationMode: 'Password',
+      password: 'abc123',
+    }
+
+    actual = customerSync.buildActions(now, before)
+    expected = [
+      {
+        action: 'setAuthenticationMode',
+        authMode: now.authenticationMode,
+        password: now.password,
+      },
+    ]
+
+    expect(actual).toEqual(expected)
+
+    before = {}
+    now = {
+      authenticationMode: 'ExternalAuth',
+    }
+
+    actual = customerSync.buildActions(now, before)
+    expected = [
+      {
+        action: 'setAuthenticationMode',
+        authMode: now.authenticationMode,
+      },
+    ]
+    expect(actual).toEqual(expected)
+
+    before = {
+      authenticationMode: 'ExternalAuth',
+    }
+    now = {}
+
+    actual = customerSync.buildActions(now, before)
+
+    expected = []
+    expect(actual).toEqual(expected)
+
+    before = {
+      authenticationMode: 'ExternalAuth',
+    }
+    now = {
+      authenticationMode: '',
+    }
```

Comment:
little bit doubtful here, cause in this case it should be unset, but seems there is no action for this.

Reply:
me too, thats why i didnt generate any action for this, to avoid any unwanted behaviour. but im also not sure